lib/ts/Makefile.am

```diff
@@ -28,7 +28,7 @@ AM_CPPFLAGS = -I$(top_srcdir)/lib
 
 lib_LTLIBRARIES = libtsutil.la
 
-libtsutil_la_LDFLAGS = -no-undefined -version-info @TS_LIBTOOL_VERSION@
+libtsutil_la_LDFLAGS = -no-undefined -version-info @TS_LIBTOOL_VERSION@ @LIBTOOL_LINK_FLAGS@
```

Comment:
Can you explain to me why this is a problem only in 6.2.x?

Reply:
Not sure I can explain this one as I'm not an expert in the build tools. It looks like there was significant re-working between 6.2.x and master since post-configure output is as follows --

For 6.2.x --
```
configure: Build option summary:
    CC:                 cc
    CXX:                c++
    CPP:                cc -E
    CFLAGS:             -I/home/pebc/local/include -std=gnu99 -g -pipe -Wall -O3 -feliminate-unused-debug-symbols -fno-strict-aliasing -mcx16
    CXXFLAGS:            -std=c++11 -g -pipe -Wall -O3 -feliminate-unused-debug-symbols -fno-strict-aliasing -Wno-invalid-offsetof -mcx16
    CPPFLAGS:           -I/home/pebc/local/include -Dlinux -D_LARGEFILE64_SOURCE=1 -D_COMPILE64BIT_SOURCE=1 -D_GNU_SOURCE -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -I/home/pebc/opt/openssl/include -DOPENSSL_NO_SSL_INTERN -I/usr/include/tcl8.6 -I/usr/include/libxml2
    LDFLAGS:            -L/home/pebc/local/lib -L/home/pebc/opt/openssl/lib
    SHARED_CFLAGS:      -fPIC
    SHARED_CXXFLAGS:    -fPIC
    SHARED_CXXLINKFLAGS:
    SHARED_LDFLAGS:     -shared
    OPENSSL_LDFLAGS:    -L/home/pebc/opt/openssl/lib
    OPENSSL_INCLUDES:   -I/home/pebc/opt/openssl/include
    EXTRA_CC_LDFLAGS:
    EXTRA_CXX_LDFLAGS:  -rdynamic
    LIBTOOL_LINK_FLAGS: -R/home/pebc/opt/openssl/lib
```
For master --
```
configure: Build option summary:
    CC:                 gcc
    CXX:                c++
    CPP:                gcc -E
    AM_CFLAGS:          -std=gnu99 -g -pipe -Wall -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -Wno-format-truncation -O3 -feliminate-unused-debug-symbols -fno-strict-aliasing -mcx16
    AM_CXXFLAGS:        -std=c++11 -g -pipe -Wall -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -Wno-format-truncation -O3 -feliminate-unused-debug-symbols -fno-strict-aliasing -Wno-invalid-offsetof -mcx16
    AM_CPPFLAGS:        -Dlinux -D_LARGEFILE64_SOURCE=1 -D_COMPILE64BIT_SOURCE=1 -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -I/usr/include/tcl8.6
    AM_LDFLAGS:         -rdynamic -Wl,--as-needed -R/home/pebc/opt/openssl/lib
    SHARED_CFLAGS:      -fPIC
    SHARED_CXXFLAGS:    -fPIC
    SHARED_CXXLINKFLAGS:
    SHARED_LDFLAGS:     -shared
    OPENSSL_LDFLAGS:    -L/home/pebc/opt/openssl/lib
    OPENSSL_INCLUDES:   -I/home/pebc/opt/openssl/include
    LUAJIT_CFLAGS:      -I/home/pebc/local/include
    LUAJIT_CPPFLAGS:    -I$(abs_top_srcdir)/lib/luajit/src
    LUAJIT_LDFLAGS:
```
So the custom OpenSSL directory is included in LIBTOOL_LINK_FLAGS in 6.2.x and is included by AM_LDFLAGS in master. So perhaps it is passed to the linker under master via the AM_LDFLAGS variable while it was not passed to the linker via LIBTOOL_LINK_FLAGS in 6.2.x without the change  in this PR.

Reply:
Additional info -- even under master a 'ldd' of 'libtsutil.so' will still show 'libcrypto.so.1.1' as 'not found'. However, it ends up working because the linker, when creating test_Huffmancode under proxy/http2 knows about the custom OpenSSL directory due to the AM_LDFLAGS.

For master --
```
$ egrep AM_LDFLAGS proxy/http2/Makefile
        $(CXXFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
        $(AM_LDFLAGS) $(LDFLAGS) -o $@
AM_LDFLAGS = -rdynamic -Wl,--as-needed -R/home/pebc/opt/openssl/lib
```
For 6.2.x --
```
$ egrep AM_LDFLAGS proxy/http2/Makefile
        $(CXXFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
        $(AM_LDFLAGS) $(LDFLAGS) -o $@
```
So it seems the -R<custom-ssl-directory> can be given when creating the libtsutil.so (which is what is done in this PR) or when a dependent object is created (which is what is done under master).